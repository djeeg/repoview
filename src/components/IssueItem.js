import React, { Component } from 'react';

//todo: could be a stateless component
class IssueItem extends Component {

    constructor(props, context) {
        super(props, context);

        //this.state = {};
    }

  render() {
    return (
      <li style={{color: 'green'}}>
          {this.props.title}
      </li>
    );
  }
}

export default IssueItem;
