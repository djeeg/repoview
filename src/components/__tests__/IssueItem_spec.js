import * as React from "react";
const ReactShallowRenderer = require("react-test-renderer/shallow");
const ReactTestRenderer = require("react-test-renderer");

import IssueItem from "../IssueItem";

describe("<IssueItem />", () => {
    it("renders", () => {
        const tree = ReactTestRenderer.create(<IssueItem />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("renders with title", () => {
        const tree = ReactTestRenderer.create(<IssueItem title={"Hello"} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("renders shallow", () => {
        const renderer = new ReactShallowRenderer();
        renderer.render(<IssueItem />);
        const result = renderer.getRenderOutput();
        expect(result.type).toBe("li");
    });
});
