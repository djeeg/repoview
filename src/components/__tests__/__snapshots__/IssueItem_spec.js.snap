// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IssueItem /> renders 1`] = `
<li
  style={
    Object {
      "color": "green",
    }
  }
/>
`;

exports[`<IssueItem /> renders with title 1`] = `
<li
  style={
    Object {
      "color": "green",
    }
  }
>
  Hello
</li>
`;
